### PR TITLE
m3dacfix

### PR DESCRIPTION
**Pull request by thattil**
m3 server: 
- fix vth dac settings
  - setting counter mask changes vth dac values (disabling sets to 2800)
  - vthrehsold only changes for enabled counters
  - setting vth overwrites dac even if counter disabled
  - counters enabled, sets remembered values
  - added documentation
- isMaster used absolute address, fixed that
- - only tested with virtual server
@anberga

### CODE REVIEW

File: slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c [hunk 1]
```diff
@@ -938,8 +936,9 @@ void setCounterMask(uint32_t arg) {
     for (int i = 0; i < NCOUNTERS; ++i) {
         // if change in enable
         if ((arg & (1 << i)) ^ (oldmask & (1 << i))) {
-            // will disable if counter disabled
-            setDAC(VTH1, vthEnabledVals[i], 0);
+            // will disable if counter disabled, else set corresponding vth dac
+            enum DACINDEX vth = (i == 0 ? M_VTH1 : (i == 1 ? M_VTH2 : M_VTH3));
```
Comment (erikfrojdh) [inline slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c:940]: The double ternary operator took me a few seconds to understand =) 

maybe make something like 
DACINDEX ind[3]{M_VTH1, M_VTH2, M_VTH3}
for ...
   ind[i]

**Comment by erikfrojdh** [comment]
Looks good, I would consider the array change for clarity (and maybe it's even better for the compiler) 
otherwise go ahead after testing

File: slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c (hunk 1 above)
Comment (thattil) [reply slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c:940]: makes sense!

**Comment by thattil** [comment]
> Looks good, I would consider the array change for clarity (and maybe it's even better for the compiler)
> otherwise go ahead after testing

alright.. will merge after anna tests

File: slsDetectorServers/slsDetectorServer/include/slsDetectorFunctionList.h [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #include "blackfin.h"
 #endif
 
-#if defined(MYTHEN3D) 
+#if defined(MYTHEN3D)
```
Comment (erikfrojdh) [inline slsDetectorServers/slsDetectorServer/include/slsDetectorFunctionList.h:28]: do we do #ifdef or if defined? See line 320 =)

File: slsDetectorSoftware/tests/test-CmdProxy-mythen3.cpp [hunk 3]
```diff
@@ -49,69 +49,125 @@ TEST_CASE("Setting and reading back MYTHEN3 dacs", "[.cmd][.dacs]") {
...
                 REQUIRE(oss.str() == "dac vthreshold 1234\n");
             }
             {
                 std::ostringstream oss;
-                proxy.Call("vthreshold", {}, -1, GET, oss);
+                proxy.Call("dac", {"vthreshold"}, -1, GET, oss);
                 REQUIRE(oss.str() == "dac vthreshold 1234\n");
             }
 
+            // disabling counters change vth values
```
Comment (thattil) [inline slsDetectorSoftware/tests/test-CmdProxy-mythen3.cpp:65]: Need to add documentation in help

File: slsDetectorServers/slsDetectorServer/include/slsDetectorFunctionList.h (hunk 2 above)
Comment (thattil) [reply slsDetectorServers/slsDetectorServer/include/slsDetectorFunctionList.h]: done

File: slsDetectorServers/slsDetectorServer/src/slsDetectorServer_funcs.c [hunk 4]
```diff
@@ -7576,21 +7576,23 @@ int set_pattern(int file_des) {
...
-	/****************************************************************************************************************/
-	/* I SUGGEST TO VALIDATE THE VALUES HERE AND THEN WRITE THE PATTERN IN A SEPARATE FUNCTION WHICH COULD BE REUSED*/
-	/* added loadPattern.c/h - the same func could be reused also in readDefaultPattern                                  */
-	/***************************************************************************************************************/
+            ("Setting Pattern Word (printing every 10 words that are not 0\n"));
+        /****************************************************************************************************************/
+        /* I SUGGEST TO VALIDATE THE VALUES HERE AND THEN WRITE THE PATTERN IN A
+         * SEPARATE FUNCTION WHICH COULD BE REUSED*/
+        /* added loadPattern.c/h - the same func could be reused also in
+         * readDefaultPattern                                  */
```
[the remark this answers is not in the document]
Comment (thattil) [reply slsDetectorServers/slsDetectorServer/src/slsDetectorServer_funcs.c:7594]: postponed

**Pull request merged**